src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -408,10 +408,23 @@ class FeatureToggleService {
         }
 
         const constraintValuesLimit = this.resourceLimits.constraintValues;
+        const coompareWithExistingValues =
+            constraints.existing.length === constraints.updated.length;
         const constraintOverLimit = constraints.updated.find(
-            (constraint) =>
-                Array.isArray(constraint.values) &&
-                constraint.values?.length > constraintValuesLimit,
+            (constraint, i) => {
+                const updatedConstraintValueCount =
+                    constraint.values?.length ?? 0;
+                const overLimit =
+                    Array.isArray(constraint.values) &&
+                    updatedConstraintValueCount > constraintValuesLimit;
+                if (!overLimit) return false;
+                const allowAnyway = coompareWithExistingValues
+                    ? (constraints.existing[i].values?.length ?? 0) >=
+                      updatedConstraintValueCount
+                    : false;
+
+                return !allowAnyway;
+            },
```

Comment:
This could also be something like this, if you think that's more legible:

```ts
        const allowLimitExceeded =
            constraints.existing.length === constraints.updated.length
                ? (count, index) =>
                      (constraints.existing[index].values?.length ?? 0) >= count
                : () => false;
        const constraintOverLimit = constraints.updated.find(
            (constraint, i) => {
                const updatedConstraintValueCount =
                    constraint.values?.length ?? 0;
                const overLimit =
                    Array.isArray(constraint.values) &&
                    updatedConstraintValueCount > constraintValuesLimit;
                return (
                    overLimit &&
                    !allowLimitExceeded(updatedConstraintValueCount, i)
                );
            },
```

It's a little complex anyway, but 🤷🏼

Reply:
wdyt?
```ts
    private validateConstraintsLimit(constraints: {
        updated: IConstraint[];
        existing: IConstraint[];
    }) {
        if (!this.flagResolver.isEnabled('resourceLimits')) return;

        const { constraints: constraintsLimit, constraintValues: constraintValuesLimit } = this.resourceLimits;

        if (constraints.updated.length > constraintsLimit && constraints.updated.length > constraints.existing.length) {
            throwExceedsLimitError(this.eventBus, {
                resource: 'constraints',
                limit: constraintsLimit,
            });
        }

        const isSameLength = constraints.existing.length === constraints.updated.length;
        const constraintOverLimit = constraints.updated.find((constraint, i) => {
            const updatedCount = constraint.values?.length ?? 0;
            const existingCount = constraints.existing[i]?.values?.length ?? 0;

            const isOverLimit = Array.isArray(constraint.values) && updatedCount > constraintValuesLimit;
            const allowAnyway = isSameLength && existingCount >= updatedCount;

            return isOverLimit && !allowAnyway;
        });

        if (constraintOverLimit) {
            throwExceedsLimitError(this.eventBus, {
                resource: `constraint values for ${constraintOverLimit.contextName}`,
                limit: constraintValuesLimit,
                resourceNameOverride: 'constraint values',
            });
        }
    }

```

Reply:
Yeah, I like it

Reply:
a644ddd2638bb991c91bb09927f8131163e3b6e6